### ISSUE TITLE
您好，请问在一些需要计算多个输入图像转换副本的平均梯度时，代码里是否有实现计算平均梯度进行扰动更新这一操作？

### ISSUE THREAD

**Issue by lazylittlezhao** [Trustworthy-AI-Group/TransferAttack#96]
比如SIA论文里描述的需要计算平均梯度，但是我在代码中没有发现这一步骤的显示实现，我代码能力比较薄弱qwq，希望可以解答我的困惑，十分感谢。

`        for _ in range(self.epoch):
            # Obtain the output
            logits = self.get_logits(self.transform(data+delta, momentum=momentum))

            # Calculate the loss
            loss = self.get_loss(logits, label)

            # Calculate the gradients
            grad = self.get_grad(loss, delta)

            # Calculate the momentum
            momentum = self.get_momentum(grad, momentum)

            # Update adversarial perturbation
            delta = self.update_delta(delta, data, momentum, self.alpha)

        return delta.detach()`

**Comment by Zhijin-Ge** [comment]
你好，SIA里面计算多个平均梯度的代码在下面的图里。我们可以自己定义这些输入变换函数来实现我们想要的效果，其实这部分知识涉及到python中类的继承，我们在子类中重新定义了父类的函数，然后当代码调用的时候，优先会调用子类的函数实现，这个库里面还有好多类似的实现，你可以多多参考一下其他的实现。

![Image](https://github.com/user-attachments/assets/b80303c5-2d87-4463-b007-10339b3b2949)
